Sources/ProjectDescription/Config.swift

```diff
@@ -27,38 +27,23 @@
 ///     swiftVersion: "5.9.0"
 /// )
 /// ```
-public struct Config: Codable, Equatable, Sendable {
-    /// Generation options.
-    public let generationOptions: GenerationOptions
+public typealias Config = Tuist
 
-    /// Install options.
-    public let installOptions: InstallOptions
-
-    /// Set the versions of Xcode that the project is compatible with.
-    public let compatibleXcodeVersions: CompatibleXcodeVersions
-
-    /// List of `Plugin`s used to extend Tuist.
-    public let plugins: [PluginLocation]
-
-    /// Cloud configuration.
-    public let cloud: Cloud?
+public struct Tuist: Codable, Equatable, Sendable {
+    /// Configures the project Tuist will interact with.
+    /// When no project is provided, Tuist defaults to the workspace or project in the current directory.
+    public let project: TuistProject
 
     /// The full project handle such as tuist-org/tuist.
     public let fullHandle: String?
 
     /// The base URL that points to the Tuist server.
     public let url: String
 
-    /// The Swift tools versions that will be used by Tuist to fetch external dependencies.
-    /// If `nil` is passed then Tuist will use the environment’s version.
-    /// - Note: This **does not** control the `SWIFT_VERSION` build setting in regular generated projects, for this please use
-    /// `Project.settings`
-    /// or `Target.settings` as needed.
-    public let swiftVersion: Version?
-
     /// Creates a tuist configuration.
     ///
     /// - Parameters:
+    ///   - project: The project configuration.
```

Comment:
You are either missing the `project` parameter in the function, or this should not be added to doc?

Reply:
Also, should this init be deprecated?

Reply:
👍 on a deprecation notice.  if we wanted to wait a few versions to get production feedback that feels alright with me.

Reply:
Good catch. I'll remove the parameter from the docs, and add the deprecation notice.